spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/filter/NettyRoutingFilter.java

```diff
@@ -197,10 +201,11 @@ else if (response instanceof AbstractServerHttpResponse) {
 					return Mono.just(res);
 				});
 
-		if (properties.getResponseTimeout() != null) {
-			responseFlux = responseFlux.timeout(properties.getResponseTimeout(),
-					Mono.error(new TimeoutException("Response took longer than timeout: "
-							+ properties.getResponseTimeout())))
+		Duration responseTimeout = getResponseTimeout(route);
+		if (responseTimeout != null) {
```

Comment:
is it actually correct behavior? i suppose that we should have default read timeout in gateway..

Reply:
I kept original behavior as is.
But I'll add the default timeout if @spencergibb will ok with it